web_widget_x2many_2d_matrix/static/src/js/web_widget_x2many_2d_matrix.js

```diff
@@ -207,13 +206,27 @@ odoo.define('web_widget_x2many_2d_matrix.widget', function (require) {
                 this.field_label_y_axis, true);
         },
 
-        // return the class(es) the inputs should have
+        // return the class(es) inputs should have
         get_xy_value_class: function()
         {
-            var classes = 'oe_form_field oe_form_required';
-            if(this.is_numeric)
-            {
-                classes += ' oe_form_field_float';
+            var classes = 'edit o_form_input o_form_required oe_edit_only';
```

Comment:
But what if the field is not required?

Reply:
Would be easy to conditionally add the attribute based on `this.get('required')` but I'd rather not until the widget actually supports this; currently when I set the matrix to be required, it still allows being saved without any actual input.

This to me would be a separate improvement than these simple style fixes.

Reply:
OK